boards: nxp: mimxrt1170_evk & mimxrt1160_evk:  M4 improvements

DerekSnell opened the pull request:
- Fixes issue where zephyr,flash in ocram node overlaps with physical SRAM sram1 used for zephyr,sram.
- Moves default zephyr,flash from external flash to M4 ITCM to improve performance.
- Updates overlays that moved zephyr,flash to ocram now use default
- enable M4 CONFIG_NET_L2_ETHERNET

decsny commented on boards/nxp/mimxrt1160_evk/mimxrt1160_evk_mimxrt1166_cm4.dts:
```diff
@@ -38,6 +38,25 @@
+	 * The M7 core can access this OCRAM alias for M4 ITCM and copy
+	 * the M4 image here.
+	 */
+	ocram_m4_itcm: ocram_m4_itcm@20200000 {
```
I would think this belong in SOC dtsi then?

Commit:
boards: nxp: mimxrt1170_evk & rt1160: enable M4 CONFIG_NET_L2_ETHERNET

The Ethernet networking stack can run on the M4 core, see
https://community.nxp.com/t5/Zephyr-Project-Knowledge-Base/Zephyr-networking-stack-on-i-MX-RT1170-Cortex-M4-secondary-core/ta-p/2159461

Signed-off-by: Derek Snell <derek.snell@nxp.com>

Commit:
boards: nxp: mimxrt1170_evk: improve zephyr,flash location for M4

- Fixes issue where zephyr,flash in ocram node overlaps with physical
  SRAM sram1 used for zephyr,sram.
- Updates overlays that moved zephyr,flash to ocram now use default

Signed-off-by: Derek Snell <derek.snell@nxp.com>

Commit:
boards: nxp: mimxrt1160_evk: improve zephyr,flash location for M4

- Fixes issue where zephyr,flash in ocram node overlaps with physical
  SRAM sram1 used for zephyr,sram.
- Moves default zephyr,flash from external flash to M4 ITCM to improve
  performance.
- Updates overlays that moved zephyr,flash to ocram now use default

Signed-off-by: Derek Snell <derek.snell@nxp.com>
